src/main/MQ2PostOffice.h

```diff
@@ -23,19 +24,30 @@ void ShutdownPipeClient();
 
 namespace pipeclient {
 
-MQLIB_API void NotifyCharacterLoad(const char* Profile, const char* Account, const char* Server, const char* Character);
-MQLIB_API void NotifyCharacterUnload(const char* Profile, const char* Account, const char* Server, const char* Character);
-MQLIB_API void NotifyCharacterUpdate(const char* Class, const char* Level);
-MQLIB_API void LoginServer(const char* Login, const char* Pass, const char* Server);
-MQLIB_API void LoginCharacter(const char* Login, const char* Pass, const char* Server, const char* Character);
-MQLIB_API void LoginProfile(const char* Profile, const char* Server, const char* Character);
-
 MQLIB_API uint32_t GetLauncherProcessID();
 
 void NotifyIsForegroundWindow(bool isForeground);
 void RequestActivateWindow(HWND hWnd, bool sendMessage = true);
 
-} // namespace pipeclient
+MQLIB_OBJECT std::shared_ptr<mailbox::PostOffice::Mailbox> AddMailbox(const std::string& localAddress, mailbox::PostOffice::ReceiveCallback receive);
+MQLIB_OBJECT bool RemoveMailbox(const std::string& localAddress);
+
+MQLIB_OBJECT void RouteMessage(MQMessageId messageId, const void* data, size_t length);
+template <typename ID>
+void RouteMessage(ID messageId, const void* data, size_t length)
+{
+	RouteMessage(static_cast<MQMessageId>(messageId), data, length);
+}
 
+MQLIB_OBJECT void RouteMessage(MQMessageId messageId, const std::string& data);
+template <typename ID>
+void RouteMessage(ID messageId, const std::string& data)
+{
+	RouteMessage(static_cast<MQMessageId>(messageId), data);
+}
+
+MQLIB_OBJECT void RouteMessage(const std::string& data); // envelopes come as strings, and will always want "ROUTE" message IDs
```

Comment:
if this is the public api, it should be somewhere in include/mq/ and have comments describing each

Reply:
Hmm, this is actually a good point. I think I originally intended for this to be for managing mailboxes in the MQ Main post office, but I think the line is muddy now and the MQ main post office should be the only post office in MQ (the launcher would have the other post office currently, and other apps would have other post offices)

Reply:
alright, I moved all this stuff to mq/proto/PostOffice.h -- there are still some leftover functions here (and in the launcher) that directly send messages, which is legacy support. We could move those to actual actors in the future.